provider/core/src/datagen/mod.rs

```diff
@@ -108,11 +106,14 @@ macro_rules! make_exportable_provider {
                 match key.hashed() {
                     $(
                         $struct_m => {
-                            let heap_stats = from.attempt_zero_copy_heap_size::<$struct_m>();
-                            return Ok($crate::DataPayload::from_owned(heap_stats));
+                            let _reified_data: $crate::DataPayload<$struct_m> = buf
+                                .into_deserialized($crate::buf::BufferFormat::Postcard1)
+                                .expect("Failed to deserialize BufferMarker as postcard-1.0");
+                            use $crate::dynutil::UpcastDataPayload;
+                            Ok($crate::AnyMarker::upcast(_reified_data))
```

Comment:
We only need the side effect of this, but I thought we might as well return it as `Any`

Reply:
Ah this doesn't work because the upcast allocates